lib/interpreters/compute_several_solutions.rb

```diff
@@ -238,37 +238,47 @@ def self.find_best_heuristic(service_vrp)
     end
 
     def self.select_best_heuristic(vrp)
-      vehicles = vrp[:vehicles] || []
-      services = vrp[:services] || []
-      shipments = vrp[:shipments] || []
+      vehicles = vrp.vehicles
+      services = vrp.services
 
       loop_route = vehicles.any?{ |vehicle|
-        if (vehicle.start_point_id.nil? || vehicle.end_point_id.nil?) && (vehicle.start_point.nil? || vehicle.end_point.nil?)
+        if (vehicle.start_point_id.nil? || vehicle.end_point_id.nil?) &&
+           (vehicle.start_point.nil? || vehicle.end_point.nil?)
           true
         else
           start_point = vehicle.start_point
           end_point = vehicle.end_point
 
           vehicle.start_point_id == vehicle.end_point_id ||
-            start_point[:location] && end_point[:location] && start_point[:location][:lat] == end_point[:location][:lat] && start_point[:location][:lon] == end_point[:location][:lon] ||
-            vrp[:matrices] && start_point[:matrix_index] && end_point[:matrix_index] && vrp[:matrices].all?{ |matrix| matrix[:time] && matrix[:time][start_point[:matrix_index]][end_point[:matrix_index]] == 0 }
+            start_point.same_location?(end_point) ||
+            vrp.matrices && start_point.matrix_index && end_point.matrix_index &&
+              vrp.matrices.all?{ |matrix|
+                matrix.time && matrix.time[start_point.matrix_index][end_point.matrix_index] == 0
+              }
         end
       }
-      size_mtws = services.select{ |service| service[:timewindows].to_a.size > 1 }.size
-      size_rest = vehicles.collect{ |vehicle| vehicle[:rests].to_a.size }.sum
-      unique_configuration = vehicles.collect{ |vehicle| vehicle[:router_mode] }.uniq.size == 1 && vehicles.collect{ |vehicle| vehicle[:router_dimension] }.uniq.size == 1 &&
-                             vehicles.collect{ |vehicle| vehicle[:start_point_id] }.uniq.size == 1 && vehicles.collect{ |vehicle| vehicle[:end_point_id] }.uniq.size == 1
-
-      if vehicles.any?{ |vehicle| vehicle[:overall_duration] }
+      size_mtws = services.count{ |service| service.activity.timewindows.size > 1 }
+      size_rest = vehicles.map{ |vehicle| vehicle.rests.size }.sum
```

Comment:
I am not 100% sure but the following should work
```suggestion
      size_rest = vehicles.sum{ |vehicle| vehicle.rests.size }
```